.github/workflows/ci.yml

```diff
@@ -11,8 +11,8 @@ env:
   # This should be the same version of Rust which the above stable was branched from.
   # To get the correct date, use `cargo --version`. E.g for 1.77, this prints
   # cargo 1.77.0 (3fe68eabf 2024-02-29)
-  # So the date used when we depended on 1.77 was 2024-02-29
-  RUST_DOCS_COMPILE_VER: "nightly-2024-02-29"
+  # So the date used when we depended on 1.77 was 2024-02-29.
+  RUST_DOCS_COMPILE_VER: "nightly-2024-06-03"
```

Comment:
We probably should bump the stable version as well

Reply:
Yeah, I'll eventually be doing a pass on all the repos to make sure the versions are bumped. I'm omitting that in this script sync pass to reduce PR size.